overlord/managers_test.go

```diff
@@ -2086,12 +2086,15 @@ type: kernel`
 	c.Assert(enabledTryKernels, HasLen, 1)
 	c.Assert(enabledTryKernels[0].Filename(), Equals, kernelSnapInfo.Filename())
```

Comment:
I didn't notice this before, sorry, but these tests should probably somehow check more for state (what kernel do we have in boot, what try-kernel if one) instead of what we did, it feels a bit the wrong level. The churn is a bit of sign, my wondering above are a sign as well

Reply:
I assume you mean checking the kernel set in the mock bootloader? Yes we can do this, it is a good check to have here. 

In general though, I would like to have more tests in managers here for the different failover cases, but it becomes quite repetitive and ends up being a lot to write and review. What I would kind of like to do instead is to take the intitramfs_test.go I have in my other PR, and extract out a helper for MarkBootSuccessful for just one type of snap, then we can setup the various scenarios, run the initramfs logic, then run the user-space snapd logic with MarkBootSuccessful and then check for cleanups at the end. That way we can minimally test all the cleanup logic with a more real system